rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2620,6 +2620,80 @@ void voteFederationChange_addFederatorMultiKeyWithDifferentMstKey_returnsSuccess
 
         }
 
+        private Stream<Arguments> federatorsWithDifferentKeysButOneParametersProvider() {
+
+            BtcECKey federator1BtcKey = BtcECKey.fromPrivate(BigInteger.valueOf(100));
+            ECKey federator1RskKey = ECKey.fromPrivate(BigInteger.valueOf(200));
+            ECKey federator1MstKey = ECKey.fromPrivate(BigInteger.valueOf(300));
+
+            BtcECKey federator2BtcKey = BtcECKey.fromPrivate(BigInteger.valueOf(400));
+            ECKey federator2RskKey = ECKey.fromPrivate(BigInteger.valueOf(500));
+            ECKey federator2MstKey = ECKey.fromPrivate(BigInteger.valueOf(600));
+
+            return Stream.of(
+                // 2 federators with different btc and rsk keys and same mst key
+                Arguments.of(federator1BtcKey, federator1RskKey, federator1MstKey, federator2BtcKey, federator2RskKey, federator1MstKey),
+                // 2 federators with different btc and mst keys and same rsk key
+                Arguments.of(federator1BtcKey, federator1RskKey, federator1MstKey, federator2BtcKey, federator1RskKey, federator2MstKey),
+                // 2 federators with different rsk and mst keys and same btc key
+                Arguments.of(federator1BtcKey, federator1RskKey, federator1MstKey, federator1BtcKey, federator2RskKey, federator2MstKey)
+            );
+        }
+
+        @ParameterizedTest
+        @MethodSource("federatorsWithDifferentKeysButOneParametersProvider")
+        void voteFederationChange_add2FederatorsWithWithDifferentKeysButOne_returnsFederatorAlreadyPresentResponseCode(
+            BtcECKey federator1BtcKey,
+            ECKey federator1RskKey,
+            ECKey federator1MstKey,
+            BtcECKey federator2BtcKey,
+            ECKey federator2RskKey,
+            ECKey federator2MstKey
+        ) {
+
+            Transaction firstAuthorizedTx = TransactionUtils.getTransactionFromCaller(signatureCache, FederationChangeCaller.FIRST_AUTHORIZED.getRskAddress());
+            Transaction secondAuthorizedTx = TransactionUtils.getTransactionFromCaller(signatureCache, FederationChangeCaller.SECOND_AUTHORIZED.getRskAddress());
+
+            ABICallSpec createFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.CREATE.getKey(), new byte[][]{});
+
+            // Voting with m of n authorizers to create the pending federation
+            federationSupport.voteFederationChange(firstAuthorizedTx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(secondAuthorizedTx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            ABICallSpec addMultiKeyAbiCallSpec1 = new ABICallSpec(FederationChangeFunction.ADD_MULTI.getKey(), new byte[][]{
+                federator1BtcKey.getPubKey(),
+                federator1RskKey.getPubKey(),
+                federator1MstKey.getPubKey(),
+            });
```

Comment:
You could create a helper method to create the byte[][] with the keys. Wdyt?

Reply:
Updated.